validator/src/main.rs

```diff
@@ -189,7 +189,10 @@ fn get_rpc_addr(
     identity_keypair: &Arc<Keypair>,
     entrypoint_gossip: &SocketAddr,
 ) -> (Pubkey, SocketAddr) {
-    let mut cluster_info = ClusterInfo::new(node.info.clone(), identity_keypair.clone());
+    let mut cluster_info = ClusterInfo::new(
+        ClusterInfo::spy_contact_info(&identity_keypair.pubkey()),
```

Comment:
By the way, [this change](https://github.com/solana-labs/solana/commit/e6af4511a8ad22eeba73339ddee54dc05931e5a6#diff-a8d11c36b97d52f7aa49ba42e46c8c75R625) already prevents this premature supermajority as a side effect... ;)

Reply:
So this PR is rather correctness improvement PR..

Reply:
It also helps avoid advertising an RPC or TVU port that is not actually usable, so it's a nice thing to do for our fellow cluster nodes